src/main/java/org/torproject/onionoo/server/NodeIndexer.java

```diff
@@ -266,8 +268,18 @@ private void indexNodeStatuses() {
         newRelaysByVersion.get(version).add(fingerprint);
         newRelaysByVersion.get(version).add(hashedFingerprint);
       }
-      String hostName = entry.getHostName();
-      if (null != hostName) {
+      List<String> allHostNames;
+      try {
+        allHostNames = new ArrayList<>(entry.getVerifiedHostNames());
+      } catch (NullPointerException e) {
```

Comment:
Catching `NullPointerException`s is so Python. Better check for `null` first and you can easily avoid catching this exception at all. Also applies for subsequent code.